objects/src/builder/mod.rs

```diff
@@ -267,3 +270,78 @@ impl<T: Rng> AccountBuilder<T> {
         Ok(Account::new(account_id, vault, storage, account_code, self.nonce))
     }
 }
+
+pub struct NoteBuilder {
+    sender: AccountId,
+    inputs: Vec<Felt>,
+    assets: Vec<Asset>,
+    serial_num: Word,
+    tag: Felt,
+    code: String,
+    proof: Option<NoteInclusionProof>,
+}
+
+const DEFAULT_NOTE_CODE: &str = "\
+begin
+end
+";
+
+impl NoteBuilder {
+    pub fn new(sender: AccountId) -> Self {
+        Self {
+            sender,
+            inputs: vec![],
+            assets: vec![],
+            serial_num: Word::default(),
+            tag: Felt::default(),
+            code: DEFAULT_NOTE_CODE.to_string(),
+            proof: None,
+        }
+    }
+
+    pub fn note_inputs(mut self, inputs: Vec<Felt>) -> Result<Self, NoteError> {
+        NoteInputs::new(&inputs)?;
+        self.inputs = inputs;
+        Ok(self)
+    }
+
+    pub fn add_asset(mut self, asset: Asset) -> Self {
+        self.assets.push(asset);
+        self
+    }
+
+    pub fn serial_num(mut self, serial_num: Word) -> Self {
+        self.serial_num = serial_num;
+        self
+    }
```

Comment:
Is this required to be random, as it is the case with the seed for account ids? or is it suppose to be random to prevent likability?

Ideally we should try to have deterministic values were we can, so that if a test fails we can easily correlate back to the source.

Reply:
Preventing linkability is one reason, but also we need to guarantee that no two notes have the same serial number. This is needed because if two notes are identical (i.e., they have the same script, assets, inputs, serial number etc.) they will have the same nullifier, and thus, only one of them can be spent. In other words, serial number is used to make sure two notes with other identical properties have different nullifiers.

Reply:
I'm going to change this to the random value as per the request. I do want to note though that IMO this would benefit of a deterministic value, since the builder is not intended to produce data for production, but for testing. For example, the case were two notes with the same values can not be consumed, it is IMO something we definitely should test.

Reply:
Would seeding the rng with a predetermined value be suitable here?

Reply:
> Would seeding the rng with a predetermined value be suitable here?

I think this can work, yes. Another alternative is just to `copy` the original note object.